### ISSUE TITLE
Siri command usually take 2 tries after period of inactivity

### ISSUE THREAD

**Issue by hepcat72** [homebridge/homebridge#2223]
### Expected Behavior
I tell Siri "turn <on|off> X -> X should turn <on|off> (on the first try).

### Current Behavior
I tell Siri to "turn <off|on> X and 1 of 4 things happens:

1. X doesn't turn <off|on> and Siri tells me that it did turn <off|on>
2. X doesn't turn <off|on> and Siri tells me there was no response
3. X does turn <off|on> and Siri tells me there was no response
4. X does turn <off|on> and Siri tells me that it did turn <off|on>

I repeat the command and it usually works, but sometimes the above behavior is repeated and it takes a few tries.

Once it has worked correctly once, subsequent commands tend to work.  Then after a period of inactivity, the same process above repeats.

### Steps to Reproduce (for bugs)
Issue a command when I wake up in the morning or after I get home from work, or after not having issued any commands in an hour or 2.

### Versions
- **Homebridge**: 0.4.46
- **Node**:       v11.12.0
- **OS**:         Linux raspberrypi 4.14.98-v7+ #1200 SMP Tue Feb 12 20:27:48 GMT 2019 armv7l GNU/Linux

### Configuration
```json
{
    "bridge": {
        "name": "Homebridge",
        "username": "***",
        "port": ***,
        "pin": "***"
    },

    "description": "Homebridge config file for describing devices Siri will control.",

    "accessories": [
        {
            "accessory": "WeMo",
            "name": "Air Filter",
            "description": "The air filter in the master bedroom",
            "wemo_name": "AirFilter"
        },
        {
            "accessory": "WeMo",
            "name": "Space Heater",
            "description": "The space heater in the master bathroom",
            "wemo_name": "Bathroom AC"
        }
    ],

    "platforms": [
        {
            "platform": "LiftMaster2",
            "username": "***",
            "password": "***"
        },
        {
            "platform": "commander",
            "name": "Etekcity",
            "commands": [
                {
                    "name" : "lights",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "patio lights",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "game cube",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "couch fan",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "bed fan",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "wicker fan",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "attic",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                },
                {
                    "name" : "porch lights",
                    "type" : "outlet",
                    "cmd" : "~/device_control.sh",
                    "no_arg" : false
                }
            ]
        },
        {
            "platform": "CEC",
            "manufacturer": "Magnavox",
            "model": "32ME303V/F7 A",
            "sources": [
                {
                    "name": "Raspberry Pi",
                    "address": "1.0.0.0"
                },
                {
                    "name": "Mac Mini",
                    "address": "2.0.0.0"
                }
            ]
        }
    ]
}
```

### Homebridge Log / Command Output
```
tail -n 100 /home/pi/.pm2/logs/homebridge-out.log
[5/16/2019, 7:19:11 PM] [Space Heater] Found 'Bathroom AC' device at 10.0.1.50
[5/16/2019, 7:19:11 PM] [LiftMaster2] [MyQ] Logged in with MyQ user ID 0
[5/16/2019, 7:19:11 PM] [Air Filter] Found 'AirFilter' device at 10.0.1.45
[5/16/2019, 7:20:03 PM] [Etekcity] lights PowerState changed to false
[5/16/2019, 7:20:05 PM] [Etekcity] lights PowerState changed to true
[5/16/2019, 8:03:27 PM] [Etekcity] lights PowerState changed to false
[5/16/2019, 9:18:22 PM] [Space Heater] Getting power state on the 'Bathroom AC'...
[5/16/2019, 9:18:22 PM] [Air Filter] Getting power state on the 'AirFilter'...
[5/16/2019, 9:18:22 PM] [Space Heater] Power state for the 'Bathroom AC' is 0
[5/16/2019, 9:18:22 PM] [Air Filter] Power state for the 'AirFilter' is 0
[5/16/2019, 9:18:22 PM] [LiftMaster2] [MyQ Garage Door Opener] Getting current state: Closed.
[5/16/2019, 9:18:25 PM] [Etekcity] bed fan PowerState changed to true
[5/17/2019, 6:39:59 AM] [Etekcity] bed fan PowerState changed to false
[5/17/2019, 8:40:41 AM] [LiftMaster2] [MyQ] Logged in with MyQ user ID 0
[5/17/2019, 8:40:42 AM] [LiftMaster2] [MyQ Garage Door Opener] State was successfully set to Open.

head -n 100 /home/pi/.pm2/logs/homebridge-out.log
[3/16/2019, 5:30:21 PM] Loaded config.json with 2 accessories and 1 platforms.
[3/16/2019, 5:30:21 PM] ---
[3/16/2019, 5:30:21 PM] Loaded plugin: homebridge-liftmaster2
[3/16/2019, 5:30:21 PM] Registering platform 'homebridge-liftmaster2.LiftMaster2'
[3/16/2019, 5:30:21 PM] ---
[3/16/2019, 5:30:22 PM] Loaded plugin: homebridge-wemo
[3/16/2019, 5:30:22 PM] Registering accessory 'homebridge-wemo.WeMo'
[3/16/2019, 5:30:22 PM] ---
[3/16/2019, 5:30:22 PM] Loading 1 platforms...
[3/16/2019, 5:30:22 PM] [LiftMaster2] Initializing LiftMaster2 platform...
[3/16/2019, 5:30:22 PM] Loading 2 accessories...
[3/16/2019, 5:30:22 PM] [Air Filter] Initializing WeMo accessory...
[3/16/2019, 5:30:22 PM] [Air Filter] Searching for WeMo device with exact name 'AirFilter'...
[3/16/2019, 5:30:22 PM] [Space Heater] Initializing WeMo accessory...
[3/16/2019, 5:30:22 PM] [Space Heater] Searching for WeMo device with exact name 'Bathroom AC'...

tail -n 100 /home/pi/.pm2/logs/homebridge-error.log
(node:9606) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
(node:15466) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
(node:21768) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
(node:711) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
RangeError [ERR_BUFFER_OUT_OF_BOUNDS]: Attempt to write outside buffer bounds
    at boundsError (internal/buffer.js:56:11)
    at Buffer.readUInt16LE (internal/buffer.js:139:5)
    at Object.layerDecrypt (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/encryption.js:86:56)
    at HAPServer._onDecrypt (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/HAPServer.js:257:33)
    at EventedHTTPServer.emit (events.js:197:13)
    at EventedHTTPServer.<anonymous> (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/eventedhttp.js:91:70)
    at EventedHTTPServerConnection.emit (events.js:197:13)
    at EventedHTTPServerConnection._onClientSocketData (/usr/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/eventedhttp.js:244:8)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
(node:14618) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
(node:709) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
(node:14765) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
```

I redacted the port number. (I don't know if that's particularly sensitive info.)  The log heads may be quite old and demonstrate resolved issues(?).

There's only 1 ERR_BUFFER_OUT_OF_BOUNDS error in the whole log.  Everything else is related to a memory issue I encountered and fixed about a week ago.  But this behavior has been persistent since I first installed homebridge back in March.

I've been wondering if my Pi is going to sleep or something...?  Homebridge doesn't appear to be crashing.  The only boos I see recently in the logs are from my system restarts and I know I've encountered this problem numerous times in the interim.

**Comment by tobekas** [comment]
Same for me... Sometimes after long inactivity or changing wifi, Homebridge/Homekit seems to ”hang“... (both Siri access and Home App). Often it takes up to 10 seconds... then everything is fine.
But I think this is a HAP communication issue between Homebridge and the iOS device. Sometimes it‘s only hanging on one device while a second iPhone is doing perfectly fine...

**Comment by tobekas** [comment]
Maybe related to #2098.

Is your PI connected both via LAN and WLAN?

**Comment by hepcat72** [comment]
No. I’m only connected via wlan. There is no Ethernet cable connected.

**Comment by hepcat72** [comment]
I can’t attest to whether a second iOS device does fine. I haven’t got a sense of the pattern, however I know that the 3 iOS devices I use all seem to be affected and I feel like (but don’t know for sure) that when one has needed multiple commands initially, all devices then become responsive, I.e. if I have the issue on one device and use then other immediately after, it does not have the same issue.

The problem is not always reproducible either. I just got home this evening and told my iPhone 6s “turn on the lights” and it worked right away. I was a little surprised. I pretty much expect to have to issue commands twice initially when I get home.

**Comment by zilluss** [comment]
I have the same problem, wifi is completely disabled on my system, so that eliminates this as a source for the issue. I'm not sure if it's related, but the bonjour entry for homebridge goes missing from time to time.

**Comment by hepcat72** [comment]
I’m curious. Where do you find a Bonjour entry for Homebridge? I don’t know where I would find that.

**Comment by zilluss** [comment]
Bonjour Browser or avahi-browse. The homebridge entry starts with __hap

**Comment by hepcat72** [comment]
I installed Discovery (formerly "Bonjour Browser").  I don't see any "__hap" entries.  I've checked using 3 different computers a number of times.  My homebridge seems to be working currently though.

**Comment by atebeta** [comment]
I have a similar issue. Homebridge is working, but the Home app loses connection to the accessories from time to time. 

I cannot see _hap._tcp via Bonjour Browser, or in Terminal using ' dns-sd -B _services._dns-sd._udp ' or on the pi itself using ' Avahi-browse -a '. 

I can see _hap._tcp on my Mac if I use ' dns-sd -B _hap._tcp  ' :

DATE: ---Thu 13 Jun 2019---
18:08:24.418  ...STARTING...
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
18:08:24.419  Add        2   7 local.               _hap._tcp.           Homebridge-CAD8

On the Pi if I use ' avahi-browse -r _hap._tcp ' : 

+   eth0 IPv4 Homebridge-CAD8                               _hap._tcp            local
=   eth0 IPv4 Homebridge-CAD8                               _hap._tcp            local
   hostname = [CC_22_3D_E3_CE_30.local]
   address = [192.168.1.20]
   port = [38753]
   txt = ["sh=dDiHOA==" "sf=1" "ci=2" "ff=0" "s#=1" "c#=42" "id=CC:22:3D:E3:CE:30" "pv=1.0" "md=Homebridge"]
        
I have disabled IP6. The Pi is connected via ethernet only to the main router. I have another router acting as a wireless access point that some of the accessories are connected to.

**Comment by hepcat72** [comment]
Still experiencing this issue. Feels less frequent after upgrade to iOS 13, but maybe I’m just getting used to it

**Comment by hepcat72** [comment]
I think I may have randomly come upon a possible cause of this issue.  Upon opening an unrelated config file to make some edits, I noticed the pi was slow to respond.  I was having trouble in vi just moving the cursor around.

While that slowness was going on, the home app had "no response" errors appearing on numerous buttons and commands to Siri needed to be issued twice at least once.

I took a look at top and the `pilight-daemon` was taking up to 50% cpu.  `top` was the next with about 11% cpu.

I wouldn't think that that usage, though higher than I think it should be, necessarily could cause the slow-down I was seeing (in vi and otherwise), it is a potential lead.  I googled pilight cpu usage and learned that this is likely due to the receiver chip processing detected RF data.  I know that when I've run pilight-debug in the past, there is lots of junk background noise, which I imagine could be responsible for pilight using so much cpu.

I tried to figure out how to deal with this to reduce cpu load, but I had some trouble trying to figure it out.  pilight is pretty deep.  There was some mention of using a filter to reduce noise, but I couldn't find any documentation on implementing a filter.  I also briefly tried to look into simply disabling the receiver, but didn't immediately find an answer before deciding I needed to set this all aside.

But I thought I would drop this note here to indicate a possible cause for this issue.
